.travis.yml

```diff
@@ -1,94 +1,98 @@
-# This file has been generated -- see https://github.com/hvr/multi-ghc-travis
+# This Travis job script has been generated by a script via
```

Comment:
> - Why is this better?
> - Why has the script changed so drastically?

Simpler and less fragile/cumbersome caching logic; the old one was a shell-script hack that would break in some cases. `new-build` now does this natively and more principled and doesn't break easily when reinstall-situations occur.

Also the new script tests the `sdist` which helps catching packaging bugs (i.e. this exposed the HsNetworkConfig.h bug).

> - Should we be utilizing new build? The plan is to replace build with new-build, so wouldn't this script remain more stable with just build?

The thing is that what is currently called `cabal build` is a discontinued feature; and I don't use it myself, and I don't have the bandwidth to care about any bug reports related to `cabal (old-)build`. All focus is currently on getting `cabal new-build` out of the door, which uses a totally different paradigm. I'm generally trying to get everybody off my old `version 1` script (which I hardly maintain anymore), in favour of the new `version 2` script which is the one that generated the `.travis.yml` script here.